tdrs-backend/tdpservice/parsers/case_consistency_validator.py

```diff
@@ -288,3 +392,72 @@ def __validate_s2_records_are_related(self):
                     num_errors += 1
 
         return num_errors
+
+    def __validate_t5_aabd_and_ssi(self):
+        print('validate t5')
+        num_errors = 0
+        is_ssp = self.program_type == 'SSP'
+
+        t5_model_name = 'M5' if is_ssp else 'T5'
+        t5_model = self.__get_model(t5_model_name)
+
+        is_state = self.stt_type == STT.EntityType.STATE
+        is_territory = self.stt_type == STT.EntityType.TERRITORY
+
+        for rpt_month_year, reporting_year_cases in self.record_schema_pairs.sorted_cases.items():
+            t5s = reporting_year_cases.get(t5_model, [])
+
+            for record, schema in t5s:
+                rec_aabd = getattr(record, 'REC_AID_TOTALLY_DISABLED')
+                rec_ssi = getattr(record, 'REC_SSI')
+                family_affiliation = getattr(record, 'FAMILY_AFFILIATION')
+                dob = getattr(record, 'DATE_OF_BIRTH')
+
+                rpt_month_year_dd = f'{rpt_month_year}01'
+                rpt_date = datetime.strptime(rpt_month_year_dd, '%Y%m%d')
+                dob_date = datetime.strptime(dob, '%Y%m%d')
+                is_adult = get_years_apart(rpt_date, dob_date) >= 19
+
+                if is_territory and is_adult and (rec_aabd != 1 and rec_aabd != 2):
+                    self.__generate_and_add_error(
+                        schema,
+                        record,
+                        field='REC_AID_TOTALLY_DISABLED',
+                        msg=(
+                            f'{t5_model_name} Adults in territories must have a valid value for 19C.'
+                        )
+                    )
+                    num_errors += 1
+                elif is_state and rec_aabd != 2:
+                    self.__generate_and_add_error(
+                        schema,
+                        record,
+                        field='REC_AID_TOTALLY_DISABLED',
+                        msg=(
+                            f'{t5_model_name} People in states should not have a value of 1.'
+                        )
+                    )
+                    num_errors += 1
+
+                if is_territory and rec_ssi != 2:
+                    self.__generate_and_add_error(
+                        schema,
+                        record,
+                        field='REC_SSI',
+                        msg=(
+                            f'{t5_model_name} People in territories must have value = 2:No for 19E.'
+                        )
+                    )
+                    num_errors += 1
+                elif is_state and family_affiliation == 1:
+                    self.__generate_and_add_error(
+                        schema,
+                        record,
+                        field='REC_SSI',
+                        msg=(
+                            f'{t5_model_name} People in states must have a valid value.'
```

Comment:
@jtimpe can we add the following? I just noticed that the other columns in the error report dont make reference to which field or row this error message is associated with. thats probably another ticket, so in the meantime, we should reference field in the error message:

`f'{t5_model_name}: People in states must have a valid value for {field}'`

Reply:
per our discussion, i've hard-coded the field names into the string for now, and created #2996 to capture a dynamic approach to including the field/friendly name